File: 1034-31-2379-minimum-recolors-to-get-k-consecutive-black-blocks/1034-31-2379-minimum-recolors-to-get-k-consecutive-black-blocks.cpp
class Solution {
public:
    int minimumRecolors(string blocks, int k) {
    
        int i = 0;
        int n = blocks.length();
        int ans = INT_MAX;
        int curr = 0 ;
        int j = 0;
        for(j;j<k;j++ ){
            if(blocks[j]=='W') curr++;
        }
        ans = min(ans,curr);
        for(j;j<n;j++){
            if(blocks[i]=='W'){
                curr--;
            }
            i++;
            if(blocks[j]=='W') curr++;
            ans = min(ans,curr);
        }
        return ans;
    }
};

File: 1034-31-2379-minimum-recolors-to-get-k-consecutive-black-blocks/1034-31-2379-minimum-recolors-to-get-k-consecutive-black-blocks.java
class Solution {
    public int minimumRecolors(String blocks, int k) {
            int i = 0;
        int n = blocks.length();

        int curr = 0 ;
        int j = 0;
        for(j=0;j<k;j++ ){
            if(blocks.charAt(j)=='W') curr++;
        }
        int ans = curr;
        for(j=k;j<n;j++){
            if(blocks.charAt(i)=='W'){
                curr--;
            }
            i++;
            if(blocks.charAt(j)=='W') curr++;
            ans = Math.min(ans,curr);
        }
        return ans;
    }
}